tools/speech_data_explorer/data_explorer.py

```diff
@@ -1,846 +1,1044 @@
-
-            if estimate_audio:
-                filepath = absolute_audio_filepath(item['audio_filepath'], data_filename)
-                signal, sr = librosa.load(path=filepath, sr=None)
-                bw = eval_bandwidth(signal, sr)
-                item['freq_bandwidth'] = int(bw)
-                item['level_db'] = 20 * np.log10(np.max(np.abs(signal)))
-            for k in item:
-                if k not in data[-1]:
-                    data[-1][k] = item[k]
-
-    vocabulary_data = [{'word': word, 'count': vocabulary[word]} for word in vocabulary]
-    if vocab is not None:
-        for item in vocabulary_data:
-            item['OOV'] = item['word'] not in vocabulary_ext
-
-    if metrics_available:
-        wer = wer_dist / wer_count * 100.0
-        cer = cer_dist / cer_count * 100.0
-        wmr = wmr_count / wer_count * 100.0
-
-        acc_sum = 0
-        for item in vocabulary_data:
-            w = item['word']
-            word_accuracy = match_vocab[w] / vocabulary[w] * 100.0
-            acc_sum += word_accuracy
-            item['accuracy'] = round(word_accuracy, 1)
-        mwa = acc_sum / len(vocabulary_data)
-
-    num_hours /= 3600.0
-
-    if not disable_caching:
-        with open(pickle_filename, 'wb') as f:
-            pickle.dump(
-                [data, wer, cer, wmr, mwa, num_hours, vocabulary_data, alphabet, metrics_available],
-                f,
-                pickle.HIGHEST_PROTOCOL,
-            )
-
-    return data, wer, cer, wmr, mwa, num_hours, vocabulary_data, alphabet, metrics_available
-
-
-# plot histogram of specified field in data list
-def plot_histogram(data, key, label):
-    fig = px.histogram(
-        data_frame=[item[key] for item in data],
-        nbins=50,
-        log_y=True,
-        labels={'value': label},
-        opacity=0.5,
-        color_discrete_sequence=['green'],
-        height=200,
-    )
-    fig.update_layout(showlegend=False, margin=dict(l=0, r=0, t=0, b=0, pad=0))
-    return fig
-
-
-def plot_word_accuracy(vocabulary_data):
-    labels = ['Unrecognized', 'Sometimes recognized', 'Always recognized']
-    counts = [0, 0, 0]
-    for word in vocabulary_data:
-        if word['accuracy'] == 0:
-            counts[0] += 1
-        elif word['accuracy'] < 100:
-            counts[1] += 1
-        else:
-            counts[2] += 1
-    colors = ['red', 'orange', 'green']
-
-    fig = go.Figure(
-        data=[
-            go.Bar(
-                x=labels,
-                y=counts,
-                marker_color=colors,
-                text=['{:.2%}'.format(count / sum(counts)) for count in counts],
-                textposition='auto',
-            )
-        ]
-    )
-    fig.update_layout(
-        showlegend=False, margin=dict(l=0, r=0, t=0, b=0, pad=0), height=200, yaxis={'title_text': '#words'}
-    )
-
-    return fig
-
-
-def absolute_audio_filepath(audio_filepath, audio_base_path):
```

Comment:
Please integrate Ante's updates back (to support audio_base_path)

Reply:
Done